iOS unit tests build broken due to link errors

Current master branch gets the following link errors building the iOS Unit Tests target. This must be a pretty recent regression, as both Pasin and I have run these tests within the last week.

The methods accessing these symbols are the CBForest Obj-C unit tests, which are compiled into the iOS unit test target because the CBForest project isn't set up to run iOS unit tests. The missing methods are all Obj-C related, i.e. they're methods on the C++ classes that are compiled when Obj-C is available (in .mm files.)

```
Undefined symbols for architecture x86_64:
  "cbforest::revidBuffer::revidBuffer(NSString*)", referenced from:
      -[VersionedDocument_Tests test00_RevIDs] in VersionedDocument_Tests.o
      stringToRev(NSString*) in VersionedDocument_Tests.o
      -[VersionedDocument_Tests test03_AddRevision] in VersionedDocument_Tests.o
  "cbforest::nsstring_slice::nsstring_slice(NSString*)", referenced from:
      -[Database_Test test02_CreateDoc] in Database_Test.o
      -[Database_Test test03_SaveDocs] in Database_Test.o
      -[Database_Test createNumberedDocs] in Database_Test.o
      -[Database_Test test04_EnumerateDocs] in Database_Test.o
      -[Database_Test test04_EnumerateDocsDescending] in Database_Test.o
      -[Database_Test test05_AbortTransaction] in Database_Test.o
      -[Database_Test test06_TransactionsThenIterate] in Database_Test.o
      ...
  "cbforest::nsstring_slice::~nsstring_slice()", referenced from:
      -[Database_Test test02_CreateDoc] in Database_Test.o
      -[Database_Test test03_SaveDocs] in Database_Test.o
      -[Database_Test createNumberedDocs] in Database_Test.o
      -[Database_Test test04_EnumerateDocs] in Database_Test.o
      -[Database_Test test04_EnumerateDocsDescending] in Database_Test.o
      -[Database_Test test05_AbortTransaction] in Database_Test.o
      -[Database_Test test06_TransactionsThenIterate] in Database_Test.o
      ...
  "cbforest::CollatableReader::readNSObject()", referenced from:
      roundTrip(objc_object*) in Collatable_Test.o
      -[Collatable_Test testRoundTripInts] in Collatable_Test.o
  "cbforest::CollatableBuilder::CollatableBuilder(objc_object*)", referenced from:
      collatableData(objc_object*) in Collatable_Test.o
  "cbforest::CollatableBuilder::operator<<(objc_object*)", referenced from:
      roundTrip(objc_object*) in Collatable_Test.o
      int compareCollated<NSNumber* __strong, NSNumber* __strong>(NSNumber* __strong, NSNumber* __strong) in Collatable_Test.o
      StringToCollatable(NSString*) in MapReduce_Test.o
  "cbforest::VersionedDocument::VersionedDocument(cbforest::KeyStore&, NSString*)", referenced from:
      -[VersionedDocument_Tests test01_Empty] in VersionedDocument_Tests.o
      -[VersionedDocument_Tests test03_AddRevision] in VersionedDocument_Tests.o
      -[VersionedDocument_Tests test04_DocType] in VersionedDocument_Tests.o
  "cbforest::revid::operator NSString*() const", referenced from:
      -[VersionedDocument_Tests test00_RevIDs] in VersionedDocument_Tests.o
ld: symbol(s) not found for architecture x86_64
```

--- a/CouchbaseLite.xcodeproj/project.pbxproj
+++ b/CouchbaseLite.xcodeproj/project.pbxproj
@@ -980,6 +980,8 @@
 		93829ECD1C6097510042189B /* libCBLJSViewCompiler.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 930FE55E1C58562B008107A0 /* libCBLJSViewCompiler.a */; };
 		93829EDA1C6097650042189B /* SystemConfiguration.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 93829ED91C6097650042189B /* SystemConfiguration.framework */; };
 		939C2F1E1C5973AB006E8C2B /* CBLRegisterJSViewCompiler.h in Copy Extras */ = {isa = PBXBuildFile; fileRef = 93EBA2641B2EC61E0006C66C /* CBLRegisterJSViewCompiler.h */; };
+		93AAD4F61D46EE470037B4F4 /* libCBLForestDBStorage.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 27E66B431A7307590091DA3F /* libCBLForestDBStorage.a */; };
+		93AAD5041D46EE4E0037B4F4 /* libCBLForestDBStorage.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 27E66B431A7307590091DA3F /* libCBLForestDBStorage.a */; };
 		93D21AF01A955480000AD9AF /* CBLCookieStorage.h in Headers */ = {isa = PBXBuildFile; fileRef = 93D21AEE1A955480000AD9AF /* CBLCookieStorage.h */; };
 		93D21AF21A958CEC000AD9AF /* CBLCookieStorage.m in Sources */ = {isa = PBXBuildFile; fileRef = 93D21AEF1A955480000AD9AF /* CBLCookieStorage.m */; };
 		93D21B021A9857A1000AD9AF /* CookieStorage_Tests.m in Sources */ = {isa = PBXBuildFile; fileRef = 93D21B011A9857A1000AD9AF /* CookieStorage_Tests.m */; };
@@ -2170,6 +2172,7 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				93AAD4F61D46EE470037B4F4 /* libCBLForestDBStorage.a in Frameworks */,
 				930FE5791C585B77008107A0 /* libCBLJSViewCompiler.a in Frameworks */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
@@ -2253,6 +2256,7 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				93AAD5041D46EE4E0037B4F4 /* libCBLForestDBStorage.a in Frameworks */,
 				93829ECD1C6097510042189B /* libCBLJSViewCompiler.a in Frameworks */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;

